lib/addons/opa-gatekeeper/index.ts

```diff
@@ -0,0 +1,72 @@
+import { ClusterAddOn, ClusterInfo } from "../../spi";
+
+/**
+ * Properties available to configure opa gatekeeper
+ */
+export interface OpaGatekeeperAddOnProps {
+    /**
+     * Create namespace for Gatekeeper controller 
+     * @default true
+     */
+    createNamespace?: boolean;
+
+    /** Default namespace
+     * @default gatekeeper-system
+     */
+    namespace?: string;
+
+    /**
+     * Helm chart version to use to install.
+     * @default 3.7.0-beta.1
+     */
+    version?: string;
+
+    /**
+     * Setting validatingWebhookFailurePolicy to True
+     * @default true 
+     */
+    disableValidatingWebhook?: boolean;
+
+
+     values?: {
+        [key: string]: any;
+    };
+}
+
+/**
+ * Defaults options for the add-on
+ */
+const defaultProps: OpaGatekeeperAddOnProps = {
+    namespace: 'gatekeeper-system',
+    version: '3.7.0-beta.1',
+    disableValidatingWebhook: false,
+    values: {}
+};
+
+export class OpaGatekeeperAddOn implements ClusterAddOn {
+    
+    readonly options: OpaGatekeeperAddOnProps;
+    
+    constructor(props?: OpaGatekeeperAddOnProps) {
+        this.options = { ...defaultProps, ...props };
+    }
+
+    deploy(clusterInfo: ClusterInfo): void {
+
+        const props = this.options;
+
+        clusterInfo.cluster.addHelmChart("opagatekeeper-addon", {
+            chart: "gatekeeper",
+            release: "gatekeeper",
+            repository: "https://open-policy-agent.github.io/gatekeeper/charts",
+            version: props.version,
+            namespace: props.namespace,
+            values: {
```

Comment:
You have declared the following options that customer can supply: values (since there a lot of customizations customers can apply) and disableValidatingWebhook. You are not using them in the helm chart. Please take a look at other add-ons that supply values and manipulate them. For example in my acm branch: https://github.com/aws-quickstart/ssp-amazon-eks/blob/feature/acm-integration/lib/addons/nginx/index.ts

Reply:
Addressed - let me know if this is the approach you were referring to